packages/react/src/components/ComboButton/index.js

```diff
@@ -0,0 +1,180 @@
+ */
+
+import React, { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import { ChevronDown } from '@carbon/icons-react';
+import { Button } from '../Button';
+import { IconButton } from '../IconButton';
+import { Menu } from '../Menu';
+
+import { useAttachedMenu } from '../../internal/useAttachedMenu';
+import { useId } from '../../internal/useId';
+import { useMergedRefs } from '../../internal/useMergedRefs';
+import { usePrefix } from '../../internal/usePrefix';
+
+const spacing = 4; // top and bottom spacing between the button and the menu. in px
+const defaultTranslations = {
+  'carbon.combo-button.additional-actions': 'Additional actions',
+};
+
+function defaultTranslateWithId(messageId) {
+  return defaultTranslations[messageId];
+}
+
+const ComboButton = React.forwardRef(function ComboButton(
+  {
+    children,
+    className,
+    disabled,
+    label,
+    onClick,
+    size = 'lg',
+    tooltipAlign,
+    translateWithId: t = defaultTranslateWithId,
+    ...rest
+  },
+  forwardRef
+) {
+  const id = useId('combobutton');
+  const prefix = usePrefix();
+
+  const containerRef = useRef(null);
+  const ref = useMergedRefs([forwardRef, containerRef]);
+  const [width, setWidth] = useState(0);
+  const {
+    open,
+    x,
+    y,
+    handleClick: hookOnClick,
+    handleMousedown: handleTriggerMousedown,
+    handleClose,
+  } = useAttachedMenu(containerRef);
+
+  function handleTriggerClick() {
+    if (containerRef.current) {
+      const { width: w } = containerRef.current.getBoundingClientRect();
+      setWidth(w);
+      hookOnClick();
+    }
+  }
+
+  function handlePrimaryActionClick(e) {
+    if (onClick) {
+      onClick(e);
+    }
+  }
+
+  const containerClasses = classNames(
+    `${prefix}--combo-button__container`,
+    `${prefix}--combo-button__container--${size}`,
+    {
+      [`${prefix}--combo-button__container--open`]: open,
+    },
+    className
+  );
+
+  const primaryActionClasses = classNames(
+    `${prefix}--combo-button__primary-action`
+  );
+  const triggerClasses = classNames(`${prefix}--combo-button__trigger`);
+
+  return (
+    <div {...rest} className={containerClasses} ref={ref}>
+      <Button
+        className={primaryActionClasses}
+        size={size}
+        disabled={disabled}
+        onClick={handlePrimaryActionClick}>
+        {label}
+      </Button>
+      <IconButton
+        className={triggerClasses}
+        label={t('carbon.combo-button.additional-actions')}
+        size={size}
+        disabled={disabled}
+        align={tooltipAlign}
+        aria-haspopup
+        aria-expanded={open}
+        onClick={handleTriggerClick}
+        onMouseDown={handleTriggerMousedown}
+        aria-owns={open ? id : null}>
+        <ChevronDown />
+      </IconButton>
+      <Menu
+        id={id}
+        // eslint-disable-next-line react/forbid-component-props
+        style={{
+          width: `${width}px`,
+        }}
```

Comment:
Using inline styles like this is a violation of content security policy which will block many teams from being able to use this component. [Here's more context](https://github.com/carbon-design-system/carbon/issues/12580), and also some examples in [the PR where inline styles were refactored](https://github.com/carbon-design-system/carbon/pull/12784) to be compliant.

Inline styles can be used, but they have to be set via individual properties on the [CSSStyleDeclaration](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration) object.

```
const ref = useRef(); // apply a ref to the `Menu`
useIsomorphicEffect(() => {
  if (width) {
    ref.current.style.width = `${width}px`;
  } else {
    ref.current.style.width = null;
  }
}, [width]);
```

It looks like this will apply to the other component added here, and we also may need to refactor Menu. Feel free to break this out into a separate issue if you'd like to not tackle this in this PR though.

Reply:
Thank you for the resources, I wasn't aware!

I just pushed an update to remove inline styling from ComboButton, MenuButton and Menu.